enterprise/frontend/src/embedding-sdk/components/private/InteractiveAdHocQuestion.tsx

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useState } from "react";
+
+import type { SdkClickActionPluginsConfig } from "embedding-sdk";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { initializeQBRaw } from "metabase/query_builder/actions";
+import { getQueryResults } from "metabase/query_builder/selectors";
+
+import { InteractiveQuestionResult } from "./InteractiveQuestionResult";
+
+interface InteractiveAdHocQuestionProps {
+  questionPath: string; // route path to load a question, e.g. /question/140-best-selling-products - for saved, or /question/xxxxxxx for ad-hoc encoded question config
+  onNavigateBack: () => void;
+
+  withTitle?: boolean;
+  height?: number;
+  plugins?: SdkClickActionPluginsConfig;
+}
+
+export const InteractiveAdHocQuestion = ({
+  questionPath,
+  onNavigateBack,
+  withTitle = true,
+  height,
+  plugins,
+}: InteractiveAdHocQuestionProps) => {
+  const dispatch = useDispatch();
+
+  const queryResults = useSelector(getQueryResults);
+
+  const [isQuestionLoading, setIsQuestionLoading] = useState(true);
+
+  const loadQuestion = async (
+    dispatch: ReturnType<typeof useDispatch>,
+    questionUrl: string,
+  ) => {
+    setIsQuestionLoading(true);
+
+    const { location, params } = getQuestionParameters(questionUrl);
+    try {
+      await dispatch(initializeQBRaw(location, params));
+    } catch (e) {
+      console.error(`Failed to get question`, e);
+      setIsQuestionLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadQuestion(dispatch, questionPath);
+  }, [dispatch, questionPath]);
+
+  useEffect(() => {
+    if (queryResults) {
+      setIsQuestionLoading(false);
+    }
+  }, [queryResults]);
+
+  return (
+    <InteractiveQuestionResult
+      isQuestionLoading={isQuestionLoading}
+      onNavigateBack={onNavigateBack}
+      height={height}
+      componentPlugins={plugins}
+      withResetButton
+      onResetButtonClick={onNavigateBack}
+      withTitle={withTitle}
+    />
+  );
+};
+
+// This generates route parameters based on provided URL path to use it QB (QueryBuilder) initialization logic, which loads question and all it needs. See "initializeQBRaw" redux action.
+export const getQuestionParameters = (questionPath: string) => {
+  const url = new URL(`http://metabase.com${questionPath}`); // we use a dummy host name to fill-in full URL
```

Comment:
```suggestion
  const url = new URL(questionPath,"http://metabase.com"); // we use a dummy host name to fill-in full URL
```
If we're just adding the base, maybe we can use the base url parameter?

Reply:
+1, I might also extract that to a constant so it jumps out a bit more:

```ts
/** we use a dummy host name to parse the question path's url. */
const DUMMY_URL_HOSTNAME = "http://metabase.com"

const url = new URL(questionPath, DUMMY_URL_HOSTNAME)
```